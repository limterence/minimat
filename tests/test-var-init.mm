/* test-var-init: variables are initialized to default values when declared */
int i;         /* global variable declaration: all types */
float f;
bool b;
matrix m;
sequence v;
string s;

void printlocals() {
  int i;
  float f;
  bool b;
  matrix m;
  sequence v;
  string s;
  printmat(m);  /* print locally-declared variables: no seg faults */
  printseq(v);
  printf("\"%s\" ",s);
  printint(i);
  printfloat(f);
  printbool(b);
  println();
}

int main() {
  printmat(m);   /* print globally-declared variables: no seg faults */
  printseq(v);
  printf("\"%s\" ",s);
  printint(i);
  printfloat(f);
  printbool(b);
  println();
  printlocals();
}
