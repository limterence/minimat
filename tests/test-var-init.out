[0 x 0 float]
[0 int]
"" 0   0.00 0 
[0 x 0 float]
[0 int]
"" 0   0.00 0 
